libs/cardano-ledger-core/src/Cardano/Ledger/DPState.hs

```diff
@@ -183,6 +186,20 @@ toDStatePair DState {..} =
   , "irwd" .= dsIRewards
   ]
 
+lookupDepositDState :: DState c -> (StakeCredential c -> Maybe Coin)
+lookupDepositDState dstate =
+  let currentRewardDeposits = RewardDeposits $ dsUnified dstate
+   in \k -> do
+        RDPair _ deposit <- UM.lookup k currentRewardDeposits
+        Just $! fromCompact deposit
+
+lookupRewardDState :: DState c -> (StakeCredential c -> Maybe Coin)
+lookupRewardDState dstate =
+  let currentRewardDeposits = RewardDeposits $ dsUnified dstate
+   in \k -> do
+        RDPair reward _ <- UM.lookup k currentRewardDeposits
+        Just $! fromCompact reward
+
```

Comment:
Predefined, reusable, partially applied functions.  Might want to mention these are supplied purposely to construct the function arguments used by the new functions. Might want to list them in the comments as well.